vault/ssorolecredentialsprovider.go

```diff
@@ -66,6 +69,23 @@ func (p *SSORoleCredentialsProvider) getRoleCredentials(ctx context.Context) (*s
 		RoleName:    aws.String(p.RoleName),
 	})
 	if err != nil {
+		if cached && p.OIDCTokenCache != nil {
```

Comment:
there's a possibility that a token is invalidated between the time that it's created, so I wonder if checking `cached` is necessary at all

Reply:
Checking `cached` ensures `getRoleCredentials` won't be stuck in an infinite recursion loop if some AWS bug was to cause `CreateToken` to return invalid or expired tokens. In these rare situations, I thought it would be safer to return an error immediately instead of possibly retrying forever.